File: desktop/src/types/vault.ts
// Vault types matching Rust models

export interface Vault {
    id: number;
    name: string;
    description: string | null;
    icon: string | null;
    color: string | null;
    created_at: string;
    updated_at: string;
}

export interface CreateVaultParams {
    name: string;
    description?: string | null;
    icon?: string | null;
    color?: string | null;
}

export interface UpdateVaultParams {
    name?: string | null;
    description?: string | null;
    icon?: string | null;
    color?: string | null;
}

File: desktop/src-tauri/src/entities/vault.rs
//! Vault entity definition.

use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "vaults")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
    pub description: Option<String>,
    pub icon: Option<String>,
    pub color: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::entry::Entity")]
    Entries,
}

impl Related<super::entry::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::Entries.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
